src/Query/QueryAccessHandler.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace Drupal\entity\Query;
+
+use Drupal\Core\Session\AccountInterface;
+
+/**
+ * Controls query access based on the generic entity permissions.
+ *
+ * @see \Drupal\entity\EntityAccessControlHandler
+ * @see \Drupal\entity\EntityPermissionProvider
+ */
+class QueryAccessHandler extends QueryAccessHandlerBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function buildEntityOwnerConditions($operation, AccountInterface $account) {
```

Comment:
This seems like a very specific method name and purpose for such a generic class name. Is this meant to be an example implementation?

Reply:
The method names were made to match the names in EntityAccessControlHandlerBase (and the child EntityAccessControlHandler / UncacheableAccessControlHandler). Since the query handlers match the access control handler logic, it felt right to have their general flow match as well.